guacamole-ext/src/main/java/org/apache/guacamole/net/auth/Connection.java

```diff
@@ -125,5 +125,17 @@
      *     connection.
      */
     public Set<String> getSharingProfileIdentifiers() throws GuacamoleException;
+    
+    /**
+     * Get the GuacamoleProxyConfiguration for this connection.
+     * 
+     * @return
+     *     The GuacamoleProxyConfiguration for this connection.
+     * 
+     * @throws GuacamoleException 
+     *     If configuration information cannot be retrieved or parsed.
+     */
+    public GuacamoleProxyConfiguration getGuacamoleProxyConfiguration()
```

Comment:
We have historically avoided making the assumption that extensions will define a means of connecting to guacd at the `Connection` level. The connection to an underlying Guacamole proxy, whether such a proxy is used at all, etc. is all kept abstract behind the `GuacamoleTunnel` returned by a call to `connect()`.

I think it would be better to continue avoiding that assumption.

If we will be providing such a getter for convenience (which is OK), there needs to be well-defined semantics allowing implementations to choose not to define it at that level (returning `null` perhaps).

Reply:
Okay, I've removed this entirely from the interfaces and abstracts.